src/DataProtection/DataProtection/src/KeyManagement/IKeyManagerWithDeletion.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.DataProtection.KeyManagement;
+
+/// <summary>
+/// The basic interface for performing key management operations.
+/// </summary>
+/// <remarks>
+/// Instantiations of this interface are expected to be thread-safe.
+/// </remarks>
+public interface IKeyManagerWithDeletion : IKeyManager
```

Comment:
I thought we approved `IDeletableKeyManager` and `IDeletableXmlRepository` to correspond with `IDeletableElement`.

Reply:
I don't remember picking names, but those seem fine.